cms/djangoapps/contentstore/views/import_export.py

```diff
@@ -332,92 +82,60 @@ def import_status_handler(request, course_key_string, filename=None):
-            unit = None
-            failed_item = None
-            parent = None
-            try:
-                failed_item = modulestore().get_item(exc.location)
-                parent_loc = modulestore().get_parent_location(failed_item.location)
-
-                if parent_loc is not None:
-                    parent = modulestore().get_item(parent_loc)
-                    if parent.location.category == 'vertical':
-                        unit = parent
-            except:  # pylint: disable=bare-except
-                # if we have a nested exception, then we'll show the more generic error message
-                pass
-
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'raw_err_msg': str(exc),
-                'failed_module': failed_item,
-                'unit': unit,
-                'edit_unit_url': reverse_usage_url("container_handler", parent.location) if parent else "",
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        except Exception as exc:
-            log.exception('There was an error exporting course %s', course_module.id)
-            return render_to_response('export.html', {
-                'context_course': course_module,
-                'in_err': True,
-                'unit': None,
-                'raw_err_msg': str(exc),
-                'course_home_url': reverse_course_url("course_handler", course_key),
-                'export_url': export_url
-            })
-        finally:
-            shutil.rmtree(root_dir / name)
-
-        wrapper = FileWrapper(export_file)
-        response = HttpResponse(wrapper, content_type='application/x-tgz')
-        response['Content-Disposition'] = 'attachment; filename=%s' % os.path.basename(export_file.name.encode('utf-8'))
-        response['Content-Length'] = os.path.getsize(export_file.name)
-        return response
+    export_url = reverse(
+        "course_import_export_handler",
+        kwargs={
+            "course_key_string": unicode(course_key),
+        }
+    ) + '?accept=application/x-tgz'
+
+    export_url += "&{0}".format(
+        urlencode({
+            "redirect": reverse_course_url(
+                'export_handler',
+                unicode(course_key)
+            )
+        })
+    )
 
-    elif 'text/html' in requested_format:
+    if error:
         return render_to_response('export.html', {
-            'context_course': course_module,
-            'export_url': export_url
+            "context_course": course_module,
+            "export_url": export_url,
+            "in_err": error,
+            "unit": unit,
+            "failed_module": failed_module,
+            "edit_unit_url":
+                reverse_usage_url("container_handler", parent.location)
+                if parent else "",
```

Comment:
I agree with this. @bdero  can you please make sure all existing comments are addressed?